lib/modules/deployment/contract_deployer.js

```diff
@@ -63,7 +63,15 @@ class ContractDeployer {
         }, (err, subRealArgs) => {
           nextEachCb(null, subRealArgs);
         });
-      } else {
+      } else if(arg.substr('.eth') === arg.length - 4) {
+        global.EmbarkJS.Names.resolve(arg, (err, address) => {
```

Comment:
It would be preferable to use an event instead of global. If the even doesn't exist currently, you can add it. (use a request here and a commandHandler in the ENS module)